### PR TITLE
[TD-6145]<feature>:tagfilter-add parser support

### PR DESCRIPTION
**Pull request by shenglian-zhou**

### CODE REVIEW

File: src/client/src/tscSQLParser.c [hunk 1]
```diff
@@ -4486,6 +4499,44 @@ static int32_t validateLikeExpr(tSqlExpr* pExpr, STableMeta* pTableMeta, int32_t
...
+      return invalidOperationMsg(msgBuf, msg2);
+    }
+
+    int errCode = 0;
+    regex_t regex;
+    char    regErrBuf[256] = {0};
+
+    const char* pattern = pRight->value.pz;
+    int cflags = REG_EXTENDED | REG_ICASE;
+    if ((errCode = regcomp(&regex, pattern, cflags)) != 0) {
```
Comment (dapan1121) [inline src/client/src/tscSQLParser.c:4529]: do we need to free regex?

File: src/util/src/tcompare.c [hunk 2]
```diff
@@ -344,6 +350,43 @@ int32_t compareStrPatternComp(const void* pLeft, const void* pRight) {
...
+    regfree(&regex);
+    free(str);
+    free(pattern);
+    return 1;
+  }
+
+  errCode = regexec(&regex, str, 0, NULL, 0);
+  if (errCode != 0 && errCode != REG_NOMATCH) {
+    regerror(errCode, &regex, msgbuf, sizeof(msgbuf));
+    uError("Failed to match %s with pattern %s, reason %s", str, pattern, msgbuf)
```
Comment (dapan1121) [inline src/util/src/tcompare.c]: 不匹配也正常吧，不需要错误日志吧

File: src/client/src/tscSQLParser.c (hunk 1 above)
Comment (shenglian-zhou) [reply src/client/src/tscSQLParser.c:4529]: It will be freed.

File: src/util/src/tcompare.c (hunk 2 above)
Comment (shenglian-zhou) [reply src/util/src/tcompare.c]: yes. I will change it to udebug.

**Pull request merged**